src/routes/movie-list-page/movie-list-page.jsx

```diff
@@ -0,0 +1,165 @@
+  searchMoviesByTitle,
+  sortMovies
+} from '../../services/services';
+import { useSearchParams, useLoaderData } from 'react-router-dom';
+import { DefaultHeader } from '../../components/default-header/default-header';
+
+const MovieContextEnum = {
+  edit: 'edit',
+  delete: 'delete'
+};
+const urlSearchParams = new URLSearchParams();
+
+const MovieListPage = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchQuery, setSearchQuery] = useState(searchParams.get('query') || '');
+  const [sortCriterion, setSortCriterion] = useState(searchParams.get('sortBy') || '');
+  const [activeGenre, setActiveGenre] = useState(searchParams.get('genre') || '');
+  const [movieList, setMovieList] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(useLoaderData());
+  const [modalAddMovie, setModalAddMovie] = useState(false);
+  const [modalDeleteMovie, setModalDeleteMovie] = useState(false);
+  const [modalEditMovie, setModalEditMovie] = useState(false);
+
+  const sortBy = ['release date', 'title'];
+  const genreList = ['all', 'Documentary', 'Comedy', 'Horror', 'Crime'];
+  const contextMenu = Object.keys(MovieContextEnum);
+
+  // onInit
+  useEffect(() => {
+    const fetchData = async () => {
+      setMovieList(await getMovies());
+    };
+    lazy(fetchData());
+  }, []);
+
+  useEffect(() => {
+    if (searchQuery) {
+      onSearch(searchQuery);
+    }
+    if (sortCriterion) {
+      onSortBy(sortCriterion);
+    }
+    if (activeGenre) {
+      onGenreSelect(activeGenre);
+    }
+  }, []);
+
+  const onSearch = async (query) => {
+    if (query) {
+      setSearchQuery(query);
+      urlSearchParams.set('query', query);
+      setSearchParams(urlSearchParams);
+
+      setMovieList(await searchMoviesByTitle(query));
+    } else {
+      setMovieList(await getMovies(query));
+    }
+  };
+  const onGenreSelect = async (genre) => {
+    if (genre !== 'all') {
+      setActiveGenre(genre);
+      urlSearchParams.set('genre', genre);
+      setSearchParams(urlSearchParams);
+
+      setMovieList(await searchMoviesByGenre(genre));
+    } else {
+      setMovieList(await getMovies());
+    }
+  };
+  const onSortBy = async (criteria) => {
+    setSortCriterion(criteria);
+    urlSearchParams.set('sortBy', criteria);
+    setSearchParams(urlSearchParams);
+
+    setMovieList(await sortMovies(criteria));
+  };
+  const onMovieDelete = () => {};
+  function onContextMenuClick(action, movie) {
+    setSelectedMovie(movie);
+    switch (action) {
+      case MovieContextEnum.delete:
+        setModalDeleteMovie(true);
+        break;
+      case MovieContextEnum.edit:
+        setModalEditMovie(true);
+    }
+  }
+  function onMovieTileClick(movie) {
+    setSelectedMovie(movie);
+  }
+
+  return (
+    <Suspense fallback={<div>loading</div>}>
+      <React.Fragment>
+        {selectedMovie ? (
+          <MovieDetails movie={selectedMovie} onClose={() => setSelectedMovie(null)}></MovieDetails>
```

Comment:
According to the task you need to decide which component to render using routes. However, in the current implementation you rely on the selected movie value